tools/db_bench_tool.cc

```diff
@@ -1376,6 +1403,171 @@ struct DBWithColumnFamilies {
   }
 };
 
+// Generate the keys according to the two term expe fit
+// given the probability model, you can transfer the random
+// access to such distribution according to the prefix and
+// key distribution of the following model:
+// f(x) = a*exp(b*x) + c*exp(d*x)
+// x is the access count of the key or the rank of
+// prefix that has the average access count of f(x);
+struct ExpKeyUnit {
+  int64_t start_access;
+  int64_t start_key;
+  int64_t access_count;
+  int64_t num;
+};
+
+struct PrefixUnit {
+  int64_t prefix_start;
+  int64_t prefix_access;
+  int64_t prefix_keys;
+  std::vector<ExpKeyUnit> access_set;
+};
+
+class GenerateTwoTermExpKeys {
+ public:
+  int64_t access_num_;
+  int64_t key_num_;
+  int64_t input_key_num_;
+  bool initiated_;
+  std::vector<ExpKeyUnit> access_set_;
+  std::vector<PrefixUnit> prefix_set_;
+
+  GenerateTwoTermExpKeys() {
+    access_num_ = FLAGS_num;
+    key_num_ = FLAGS_num;
+    input_key_num_ = FLAGS_num;
+    initiated_ = false;
+  }
+
+  ~GenerateTwoTermExpKeys() {}
+
+  Status InitiateExpAccess(const int64_t access, const int64_t total_keys,
```

Comment:
As discussed offline, I don't like the idea that key generation is `access` related. This makes it hard to reason about whether running it with different --reads keeps the same distribution. Let's brainstorm offline and see whether we can avoid this.

Reply:
The problem is, if we only based on the keyspace and the distribution to generate the keys to be accessed, only when the (--reads == the area of distribution), we can have a correct distribution. Otherwise, the distribution will never be close to the one user collected. From another point, for the same key space, we collect the trace fro 10 minutes, 1 hour, and 1 day. The distributions are very close. Only when we apply the total access to the distribution model, we can ensure such attributes.